### PR TITLE
test: add missing tests

### PR DESCRIPTION
**Pull request by matthiasn**
## Summary by CodeRabbit

* **Tests**
  * Vastly expanded test coverage across AI, agents, surveys, dashboards, charts, daily/daily-os flows, journals, tasks/checklists, syncing, storage, and UI widgets
  * Added many widget, repository, database, and property-based tests (error branches, edge cases, conversions, serialization, streams, timers, and UI interactions)
  * Improved helpers/fallbacks and numerous targeted regression/unit tests to harden behavior and edge-case handling

### CODE REVIEW

File: test/features/ai/helpers/prompt_builder_helper_test.dart [hunk 1]
```diff
@@ -1,17 +1,21 @@
...
 import 'package:lotti/features/ai/model/ai_config.dart';
 import 'package:lotti/features/ai/state/consts.dart';
 import 'package:lotti/features/ai/util/preconfigured_prompts.dart';
+import 'package:lotti/services/entities_cache_service.dart';
 import 'package:mocktail/mocktail.dart';
 
 import '../../../helpers/fallbacks.dart';
 import '../../../mocks/mocks.dart';
+import '../../../widget_test_utils.dart';
+import '../../categories/test_utils.dart';
```
[the remark this answers is not in the document]
Comment (matthiasn) [reply test/features/ai/helpers/prompt_builder_helper_test.dart:20]: Already addressed -- getIt is imported via package:lotti/get_it.dart (line 12). Verified present and analyzer-clean.

File: test/features/journal/ui/widgets/entry_details/entry_datetime_multipage_modal_test.dart [hunk 2]
```diff
@@ -0,0 +1,540 @@
...
+        await tester.pump(const Duration(milliseconds: 300));
+        expect(find.text('Date & Time Range'), findsOneWidget);
+      } else {
+        // Some modal implementations use a leading widget; skip if absent.
+        expect(find.text('Date from:'), findsWidgets);
+      }
+    });
+
+    testWidgets('Save button enabled after navigating back from picker (dates unchanged but interaction occurred)',
+        (tester) async {
```
[the remark this answers is not in the document]
Comment (matthiasn) [reply test/features/journal/ui/widgets/entry_details/entry_datetime_multipage_modal_test.dart]: Done -- renamed to 'Save button stays disabled after navigating into the picker and back without changing dates' and reworked the body to actually navigate into the picker and back via the back arrow before asserting Save stays disabled, so the name matches the behaviour.

File: test/features/journal/state/entry_controller_test.dart [hunk 3]
```diff
@@ -2411,4 +2412,353 @@ void main() {
...
+      reset(mockPersistenceLogic);
+      when(
+        () => mockJournalDb.journalEntityById(testTextEntry.meta.id),
+      ).thenAnswer((_) async => testTextEntry);
+      when(
+        () => mockJournalDb.journalEntityById(testAudioEntry.meta.id),
+      ).thenAnswer((_) async => testAudioEntry);
+    });
+
+    test('calls SpeechRepository.updateLanguage for non-audio entry', () async {
```
[the remark this answers is not in the document]
Comment (matthiasn) [reply test/features/journal/state/entry_controller_test.dart]: Done -- renamed to 'completes without calling SpeechRepository.updateLanguage for non-audio entry' to match the assertion (the test only verifies setLanguage completes for a non-audio entry).

File: test/features/journal/state/entry_controller_test.dart [hunk 4]
```diff
@@ -2411,4 +2412,353 @@ void main() {
...
+  group('focusNodeListener – state-change branch', () {
+    setUp(() {
+      when(
+        () => mockJournalDb.journalEntityById(testTextEntry.meta.id),
+      ).thenAnswer((_) async => testTextEntry);
+    });
+
+    test(
+      'emits dirty state with shouldShowEditorToolBar=true when focus gained',
+      () async {
```
[the remark this answers is not in the document]
Comment (matthiasn) [reply test/features/journal/state/entry_controller_test.dart:2684]: Done -- renamed to 'focusNodeListener leaves toolbar hidden when focus state is unchanged' and tightened the assertion so it also checks shouldShowEditorToolBar stays false after the guard early-returns. The focus-gained path is covered by the widget-based test in the next group.

File: test/features/ai/helpers/prompt_builder_helper_test.dart [hunk 5]
```diff
@@ -1331,5 +1335,545 @@ void main() {
...
+                  transcript: 'newest transcript text',
+                ),
+                const AudioTranscript(
+                  created: DateTime(2024, 3, 1),
+                  library: 'lib-c',
+                  model: 'model-c',
+                  detectedLanguage: 'en',
+                  transcript: 'middle transcript text',
+                ),
+              ],
```
[the remark this answers is not in the document]
Comment (matthiasn) [reply test/features/ai/helpers/prompt_builder_helper_test.dart:1791]: Already addressed -- const was removed from the three AudioTranscript constructors; they now use non-const DateTime. Verified analyzer-clean and the file's tests pass.

File: test/features/ai/helpers/prompt_builder_helper_test.dart [hunk 6]
```diff
@@ -1331,5 +1335,545 @@ void main() {
...
+              dateTo: DateTime(2024, 3, 15),
+              transcripts: [
+                const AudioTranscript(
+                  created: DateTime(2024, 1, 1),
+                  library: 'lib',
+                  model: 'model',
+                  detectedLanguage: 'en',
+                  transcript: '   ', // whitespace only
+                ),
+              ],
```
[the remark this answers is not in the document]
Comment (matthiasn) [reply test/features/ai/helpers/prompt_builder_helper_test.dart:1846]: Already addressed -- const was removed from this AudioTranscript constructor (non-const DateTime). Verified.

File: test/features/dashboards/ui/widgets/charts/dashboard_health_bp_chart_test.dart [hunk 7]
```diff
@@ -0,0 +1,378 @@
+import 'package:fl_chart/fl_chart.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:lotti/features/dashboards/state/health_chart_controller.dart';
+import 'package:lotti/features/dashboards/ui/widgets/charts/dashboard_health_bp_chart.dart';
+import 'package:lotti/features/dashboards/ui/widgets/charts/time_series/utils.dart';
```
[the remark this answers is not in the document]
Comment (matthiasn) [reply test/features/dashboards/ui/widgets/charts/dashboard_health_bp_chart_test.dart]: Already addressed -- the import now uses 'hide leftTitleWidgets' on dashboard_health_bp_chart.dart, resolving the ambiguity. Verified analyzer-clean.

File: test/features/journal/ui/widgets/entry_details/entry_datetime_multipage_modal_test.dart [hunk 8]
```diff
@@ -0,0 +1,540 @@
...
+          of: saveButton,
+          matching: find.byType(FilledButton),
+        ),
+      );
+      expect(
+        btn.onPressed,
+        isNull,
+        reason: 'Save disabled when no date change has occurred',
+      );
+    });
```
[the remark this answers is not in the document]
Comment (matthiasn) [reply test/features/journal/ui/widgets/entry_details/entry_datetime_multipage_modal_test.dart]: Done -- renamed to 'Save button stays disabled after navigating into the picker and back without changing dates' and reworked the body to actually navigate into the picker and back via the back arrow, then assert Save stays disabled.

File: test/features/journal/ui/widgets/entry_details/entry_datetime_multipage_modal_test.dart [hunk 9]
```diff
@@ -0,0 +1,540 @@
...
+        ),
+      );
+      await tester.pump();
+      await tester.tap(find.text('Open'));
+      await tester.pump();
+      await tester.pump(const Duration(milliseconds: 300));
+
+      // With valid dates, error warning is NOT visible.
+      expect(find.text('Invalid Date Range'), findsNothing);
+    });
```
[the remark this answers is not in the document]
Comment (matthiasn) [reply test/features/journal/ui/widgets/entry_details/entry_datetime_multipage_modal_test.dart]: Done -- renamed to 'valid date range does not show the invalid-range warning' and fixed the misleading comment. The dateTo < dateFrom case is covered by the dedicated 'invalid range: warning widget appears when dateTo < dateFrom' test further below.

File: test/features/ai/ui/unified_ai_progress_view_test.dart [hunk 10]
```diff
@@ -1042,4 +1044,588 @@ Generate a widget that renders a login form.''',
...
+      // Should show AiErrorDisplay with a retry button
+      expect(find.byType(AiErrorDisplay), findsOneWidget);
+
+      // Tap the retry button
+      final retryBtn = find.text('Retry');
+      if (retryBtn.evaluate().isNotEmpty) {
+        await tester.tap(retryBtn);
+        await tester.pump();
+        expect(triggerCalled, isTrue);
+      }
```
[the remark this answers is not in the document]
Comment (matthiasn) [reply test/features/ai/ui/unified_ai_progress_view_test.dart]: Done, with a correction -- the retry button is not labelled 'Retry'; AiErrorDisplay renders the localized aiInferenceErrorRetryButton ('Try Again') for retryable errors, which is exactly why the old conditional was always skipped (vacuous test). Removed the guard: it now asserts find.text('Try Again') findsOneWidget, taps it, and verifies triggerNewInference fires.

File: test/features/journal/ui/widgets/entry_details/entry_datetime_multipage_modal_test.dart [hunk 11]
```diff
@@ -0,0 +1,468 @@
...
+            entryId: '32ea936e-dfc6-43bd-8722-d816c35eb489',
+          ),
+          overrides: [override],
+        ),
+      );
+      await tester.pump();
+
+      await tester.tap(find.text('Open'));
+      await tester.pump();
+      await tester.pump(const Duration(milliseconds: 300));
```
[the remark this answers is not in the document]
Comment (matthiasn) [reply test/features/journal/ui/widgets/entry_details/entry_datetime_multipage_modal_test.dart]: Done -- extracted a file-level _openModal(tester, override, launcher, mediaQueryData, settle) helper plus _useWideModalView(tester), and converted all tests to use them. Per-test bodies now contain only the varying assertions/navigation.

File: test/features/journal/ui/widgets/entry_details/entry_datetime_multipage_modal_test.dart [hunk 12]
```diff
@@ -0,0 +1,468 @@
...
+      if (backButton.evaluate().isNotEmpty) {
+        await tester.tap(backButton);
+        await tester.pump();
+        await tester.pump(const Duration(milliseconds: 300));
+        expect(find.text('Date & Time Range'), findsOneWidget);
+      } else {
+        // Some modal implementations use a leading widget; skip if absent.
+        expect(find.text('Date from:'), findsWidgets);
+      }
+    });
```
[the remark this answers is not in the document]
Comment (matthiasn) [reply test/features/journal/ui/widgets/entry_details/entry_datetime_multipage_modal_test.dart]: Done, with a correction -- the nav bar back arrow is Icons.arrow_back_rounded (rendered by ModalUtils.modalSheetPage onTapBack leadingNavBarWidget), not arrow_back_ios_new, so the old finder never matched and the else-branch always ran. Removed the conditional: the test now asserts the back arrow exists, taps it on a wide-screen/dialog layout so it is hittable (the picker page transition is pumped first to release the AbsorbPointer), and asserts the range page reappears.

File: test/beamer/beamer_app_test.dart [hunk 13]
```diff
@@ -947,6 +1042,382 @@ void main() {
...
+
+        // AppScreen is still alive.
+        expect(find.text('Tasks'), findsOneWidget);
+        // The provider was built at least once (initial build on construction).
+        expect(aiSetupBuildCount, greaterThanOrEqualTo(1));
+
+        await tester.pumpWidget(const SizedBox.shrink());
+        await tester.pump();
+      },
+    );
```
[the remark this answers is not in the document]
Comment (matthiasn) [reply test/beamer/beamer_app_test.dart:1445]: Done -- the controller now seeds its first state with a real unseen release (WhatsNewContent) so hasUnseenRelease is true, then transitions to the empty (all-seen) state on a 1ms timer that is guaranteed to fire after the build future resolves (a microtask would have raced the async build completion). The test counts rebuilds inside aiSetupPromptServiceProvider build() (not the notifier factory, since invalidate re-runs build) and asserts the count increases after the transition.

File: test/features/agents/ui/agent_creation_modal_test.dart [hunk 14]
```diff
@@ -483,4 +508,277 @@ void main() {
...
+      await tester.pump();
+
+      // Emit a new list that no longer contains 'Alpha' — triggers
+      // didUpdateWidget (lines 213-218): _hoveredId ('p-a') is cleared
+      // because it is absent from the new filtered list.
+      streamController.add([
+        testInferenceProfile(id: 'p-b', name: 'Beta'),
+        testInferenceProfile(id: 'p-c', name: 'Gamma'),
+      ]);
+      await tester.pumpAndSettle();
```
[the remark this answers is not in the document]
Comment (matthiasn) [reply test/features/agents/ui/agent_creation_modal_test.dart:770]: Investigated thoroughly. You are right that the moveTo(-200,-200) clears _hoveredId via pointer-exit before the update, short-circuiting didUpdateWidget's clearing branch (it is guarded by a non-null _hoveredId). However the divider colour cannot distinguish didUpdateWidget cleanup from pointer-exit cleanup: the cleared id is by definition absent from the surviving list, so no surviving divider depends on it; keeping the pointer over the item triggers a re-hover (empirically the divider goes transparent again, because the bottom sheet re-anchors an item under the stationary pointer); and removing the hovered item disposes its MouseRegion, which fires onExit and clears the same id regardless of didUpdateWidget. So the branch is defensive and not isolable through the public render -- I left the test verifying the clean post-removal render rather than make it flaky. Happy to expose _hoveredId for a white-box test if you would prefer real branch coverage.

File: test/features/daily_os/state/daily_os_controller_test.dart [hunk 15]
```diff
@@ -1222,6 +1222,453 @@ void main() {
...
+            if (next.hasValue && !firstEmit.isCompleted) firstEmit.complete();
+          },
+          fireImmediately: true,
+        );
+
+        await firstEmit.future.timeout(const Duration(seconds: 5));
+        expect(emissions, isNotEmpty);
+
+        testContainer.dispose();
+      });
```
[the remark this answers is not in the document]
Comment (matthiasn) [reply test/features/daily_os/state/daily_os_controller_test.dart]: Done -- switched to fakeAsync with an advancing clock (withClock(async.getClock(justBeforeMidnight))) instead of Clock.fixed. The test now elapses 16s past the 15s poll delay so the next loop iteration sees day B != day A, calls invalidateSelf(), and the rebuilt provider re-emits for day B; it asserts the emission count increases after crossing midnight.

File: test/features/daily_os/state/unified_daily_os_data_controller_test.dart [hunk 16]
```diff
@@ -4133,4 +4133,633 @@ void main() {
...
+
+          // Recreate container so tearDown can dispose it safely.
+          container = ProviderContainer(
+            overrides: [
+              dayPlanRepositoryProvider.overrideWithValue(
+                mockDayPlanRepository,
+              ),
+            ],
+          );
+        },
```
[the remark this answers is not in the document]
Comment (matthiasn) [reply test/features/daily_os/state/unified_daily_os_data_controller_test.dart:4387]: Investigated. The _isDisposed early-return in _fetchAllData (290-294) is unreachable/unobservable via the public API: build() sets _isDisposed=false synchronously immediately before calling _fetchAllData, and that check is the first statement with no await before it, so build's fetch never sees disposed=true. The other callers (_refreshFromNotifications, _updateWithRunningTimer, _saveDayPlan) all discard the returned DailyOsData behind a not-disposed guard, so the empty fallback is never surfaced as state. The suggested completer approach blocks after the disposed check (which already passed as false), so it would not reach the branch either. I left the test verifying the normal load and that disposal does not throw; open to exposing the path for a white-box test if you want true coverage here.

File: test/features/daily_os/ui/widgets/daily_timeline_test.dart [hunk 17]
```diff
@@ -2379,4 +2397,354 @@ void main() {
...
+    testWidgets("shows current time indicator on today's date", (
+      tester,
+    ) async {
+      when(
+        () => mockCacheService.getCategoryById('cat-1'),
+      ).thenReturn(testCategory);
+
+      final today = DateTime(2026, 6);
+      final unifiedData = DailyOsData(
+        date: today,
```
[the remark this answers is not in the document]
Comment (matthiasn) [reply test/features/daily_os/ui/widgets/daily_timeline_test.dart]: Done -- made _isToday and _CurrentTimeIndicator read clock.now() instead of DateTime.now() (production behaviour is unchanged; clock.now() falls back to DateTime.now() when no clock is injected). The two tests now wrap their bodies in withClock(Clock.fixed(...)) -- 2026-06-01 for the not-today case and 2026-06-01 12:00 for the indicator -- so they pass deterministically regardless of the wall-clock date when the suite runs.
